### ISSUE TITLE
Problem: pending tx txHash can't be found in explorer or by getTransaction

### ISSUE THREAD

**Issue by kuoweilai** [crypto-org-chain/cronos-blockscout#59]
When subscribe to pendingTransactions, the txHash the node returns can't be found in explorer or getTransaction method.

I can get a stream of txHash like this:
0x70a696453ad0b4651fd8192dd429ed234fa73364c521d1fe74cac770b2a8f098
0x504e0a357ccae44377d8af570dabbbf358714a3e7c943ea48a1b1b9860764c2f
0xdeb42a6b690dbe8b2f42d8ca7538334277f01fce744a42e600768bfd6780d29e
0x94e42513062ce3f8d0047509259ab96b147b568b07f6c9b9280f292b3812696d
0x7134c3298743f8568b61a1c449daee3c4d810a604128d5f1269cc5a07a4fce3e
0x856b62076bddfdb57f58671a6c44515ed846099e1f800dcf4a239d857f3340a3
0xdabc19195bc08a7f6897ca8ef0bc700b5a4ea34f0128a2d2c207acddde0bae39
0xd994c79c6cae0fb35b75376b4c08b88c701bd0fda16167737e81855e2d487593
0x9a867e0fb94c399e2c8f4d2b1a570c9c1c85e0d7af2af25d362e4fb201d86586
0x8a79dd4630dccfb70a34347e7bedf0b4afd4e2b5d1d2c6ea3746a4a2f9916db3
0xb8ef35f76ee258acde166ed2bf0c5d45a5568049b29ce4b9cfea10b80a1e81e3
0x8906786d958219c56b19c3d213af15434847e32092d27be9ea8dcebfd1f886f1
0x5ebcfe1332d167b04e92a38ceb1fddf145049f08b181edb767e498df8c82dfaf     
0xa1e6978cbfe0182fc2fa38cc3437fd2679312e428e5c7af9063d1344dbca52c0
0xddab880840118c2af22bfb35f5a6c735e0ebf0691a1bbdf91a2ded1df1b9a7fa
0xdcd9c7aed3825b6a8d4a98c573a444c11d8b275693803038c8b40f17bc8cb596
0xd4e2822a66f64f0249caec10745d198d572f5029c210a2bc4330d38c07956ba1
0x543ebbd1c4f8e2688578d669a51cbcdf1af0d3c707c940e88348978c69bded87
0xeca22fd1ba39a6f82c4e858cefaa2991dcd8f6e026bb6b6f8d598de8d9aa6dc1
0xdef868e8e2ef9cbd0acd6e5fd73de6aa96aedd66e8d04119375ae53956e5a1c5    
0xe614dea9042dcb70f4582fb5797d645f25d3c9e2934e30b17b42b4cca6360b0e     
0x17092a977ddae3374be2d97323ce6ed9c254f1f4fad906a3d1182234123babaf
0x276431890da06a72b16ce01d3a60cf3aec4b25ddb795da9bf320b778d7916c73
0x8c49c5813b7c6a133b487571a701c8aeb779ab44a302f1795b16e211d787f47d
0x3718212f142b4ba12db4973cd6b068510913593d509a80829b9a5efcb13fa0fd   

but none of them is valid txHash

**Comment by kuoweilai** [comment]
I suspect this is more like the node issue since the retruned txHash isn't meaningful.
For example, we can get a few pending txHash:
0x5d326407a9c6ff6951d146de3691768b953f479bf3d610019c72369d9e7950e4
null
0xf55de30cb75dea9b31f13f59b9b9312df476b95da47eb2a57d644c59521eb007
null
0x20e3129cf4ef7a3751da2c23d9321c7a0683b915ad825391c97da6384f642812
null
0xa3df9765b377d98ffed48c438ceedc713ad5aec6a809d30b01cf55925a7faf18
null
0xe99e161d1ebe0443f4d35f9899182187d674f39d64195085a87b88965723c844
null

by:
```
    var subscription = web3.eth.subscribe('pendingTransactions', async (error, txHash) =>{
        try {
            var current_tx = await web3.eth.getTransaction(txHash);
            console.log(txHash)
            console.log(current_tx);
        } catch (err) {
            console.log(err)
        }
```
and we can see that by getTransaction() method, we always get null. We can't find anything with those txHash from explorer either

**Comment by WilliamXieCrypto** [comment]
There have 2 issues: 
1. getTransaction() method return null.
2. Explorer can not find the pending transactions.

For the issue 1: This should be the JSON RPC API issue which used for web3 lib.
For the issue 2: I've tried to review the code but nothing found yet and I tried with [other JSON RPC endpoints,](https://www.poa.network/v/master-1/for-developers/developer-resourses#json-rpc-endpoints), the explorer can find the pending transactions.

Now, @crypto-facs tracking the issue 1 we can re-test the Explorer side after fixing the getTransaction() method

**Comment by crypto-facs** [comment]
sorry for the late response. This txs are not meant to be found with `getTransaction()` since they have not been submitted yet. The pending transactions on cronos are taken from the node's mempool and while at that point are not yet commited to any blocks.

**Comment by kuoweilai** [comment]
Hi @crypto-facs, this is not true. In all other EVMs, ```getTransaction()``` can retrieve the tx information (such as gas price, gas limit, data) even though those tx aren't included in a block yet. There must be some bugs in the txHash encoding, since those txHash aren't reflecting any useful information.

and this is the expected behavior:
https://support.chainstack.com/hc/en-us/articles/900003426246-Subscribing-to-global-new-pending-transactions-with-web3-js

**Comment by crypto-facs** [comment]
Sorry let me rephrase this. On cronos, being built on top of Cosmos, the concept of pending transactions do not act the same way as they do on other evm chains. The closest thing there is is the [mempool](https://docs.cosmos.network/master/basics/tx-lifecycle.html#addition-to-mempool), but this is unique to each node. This is where this pending transactions are comming from for the specific node you are subscribed to. Because of this, when you query `getTransaction()` this transactions are not yet found.

Usign the mempool could be confusing, so it is a question right now wether to continue using it to support pendingTransactions functionality or just do not return anything given that there is not currently a global source of pending transactions. Sorry for the confusion 🙏

**Comment by kuoweilai** [comment]
No worries. Now I understand much better. Thank you very much for your time!

**Comment by crypto-facs** [comment]
This was good insight for us though, we are trying to find a way to make this have a more similar behaviour to other evm chains. I will keep you posted!

**Comment by ogipogi** [comment]
Hi, will this behavior changed with introduction of eip1559?

https://medium.com/cronos-chain/network-upgrade-on-cronos-testnet-v0-7-0-84e962f10e47

Thx for checking 😊

**Comment by HenryHuYu** [comment]
> Sorry let me rephrase this. On cronos, being built on top of Cosmos, the concept of pending transactions do not act the same way as they do on other evm chains. The closest thing there is is the [mempool](https://docs.cosmos.network/master/basics/tx-lifecycle.html#addition-to-mempool), but this is unique to each node. This is where this pending transactions are comming from for the specific node you are subscribed to. Because of this, when you query `getTransaction()` this transactions are not yet found.
> 
> Usign the mempool could be confusing, so it is a question right now wether to continue using it to support pendingTransactions functionality or just do not return anything given that there is not currently a global source of pending transactions. Sorry for the confusion pray

Thx for your reply. I am still confused about the following questions:
1. To my understanding, each node's mempool will be boardcast to the whole chain. but you said it is unique. could you explain it more specifically?
2. So, you mean that there is no way to get the pending transaction in the chain?
3. Is there any doc on this. I check the https://cronos.org/docs/getting-started/ but found not much helpful thing.
